examples/multi_python_versions/libs/my_310_only_lib/BUILD.bazel

```diff
@@ -0,0 +1,11 @@
+load("@pypi_3_10_only//:requirements.bzl", "requirement")
+load("@rules_python//python:py_library.bzl", "py_library")
+
+py_library(
+    name = "my_310_only_lib",
+    srcs = ["__init__.py"],
+    visibility = ["@//tests:__pkg__"],
+    deps = [
+        requirement("flask")
+    ],
```

Comment:
Just to write here what I wrote in a related thread on Slack. I think what is happening here is:
* The default python toolchain version is `3.11` and if you build any target in the repo that is not using transitions, `//python/version:python_version` will just use that.
* This means that the `flask` here will be evaluated and will fail because there is no set of dependencies for `3.11` in the deps list.

There are ways to work around it:
* Set the `target_compatible_with` for this target, see the suggestion below.
* Set `tags = ["manual"]` and create a separate target that is transitioning the `my_310_only_lib` to something a build configuration where `python_version = 3.10`.
* Add a `pip.parse` entry for the default python toolchain version.

```suggestion
    ],
    target_compatible_with = select({
        "@rules_python//python/config_settings:is_python_3.10": [],
        "//conditions:default": ["@platforms//:incompatible"],
    })
```

Not sure exactly how `rules_python` can help here best. The error allows to ensure that the packages are getting the versions that they expect, but I see how this might be adding unnecessary work on the consumer side.

Maybe there should also be a transition `py_library` implementation?

Reply:
I am not sure if it is the right choice but I did look for a version specific implementation of py_library to import when I first hit this and didn't see one. I also think just adding it in here at least for me as an adequate solution and is easy to point people to in the future. After I saw py_library didn't exist for the given version I looked through the examples here to see if someone had already shown how this could be done.

Reply:
The more I think about it the more I am convinced that users have one of 3 options available to them and this example should use that:
* Set the target_compatible_with for this target, see the suggestion below.
* Set `tags = ["manual"]` and create a separate target that is transitioning the my_310_only_lib to something a build configuration where python_version = 3.10.
* Make the toolchain version that is used in `py_library` instances like this the default, or add a pip.parse entry for the default python toolchain version.

I don't think there is anything wrong in this case with `rules_python`, so I am interested in hearing your feedback as to what makes the above options not feasible in your case.

Reply:
Sorry for the miscommunication I think the use of `target_compatible_with` is a perfectly reasonable solution and I already made this change in our internal repo and things are working as expected.

The only thing I am running into now is figuring out how to make this work with the WORKSPACE based build in this repo. We just use Bazel modules internally so I didn't have to mess with that.